CHANGELOG.md

```diff
@@ -42,10 +42,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   Older binaries may not warn user when attempting to `git push` commits
   with such signatures.
 
-* In revsets, the working-copy symbols (such as `@` and `workspace_id@`) can
-  no longer be quoted and will not be resolved as a revset aliases or revset 
-  function parameter. For example, `author(foo@)` is now an error, and
-  a revset alias `'revset-aliases.foo@' = '@'` will be ignored. 
+* In revsets, the working-copy or remote symbols (such as `@`, `workspace_id@`,
+  and `branch@remote`) can no longer be quoted as a unit. If a workspace or
+  branch name contains whitespace, quote the name like `"branch name"@remote`.
+  Also, these symbols will not be resolved as a revset aliases or revset
```

Comment:
(Sorry, I don't have much context on this but was looking at the changelog and had a question.)

I have '@_' = '@-' configured as a revset alias -- does this mean it won't work anymore when I update jj (looks like #2114 started ignoring it)?

---

While I'm here, "as a revset aliases or revset function parameter" should probably be edited to "as revset aliases or revset function parameters" and "a revset alias `'revset-aliases.foo@' = '@'`" to "the revset alias `'revset-aliases.foo@' = '@'`"?

Reply:
> I have '@_' = '@-' configured as a revset alias -- does this mean it won't work anymore when I update jj (looks like #2114 started ignoring it)?

Correct. `@_` will be parse error if this PR gets merged. To be precise, `@_` isn't ignored by #2114, but `_@` is. This PR makes these unusable as an identifier, and `@whatever` is syntax error.

> While I'm here, "as a revset aliases or revset function parameter" should probably be edited to "as revset aliases or revset function parameters" and "a revset alias `'revset-aliases.foo@' = '@'`" to "the revset alias `'revset-aliases.foo@' = '@'`"?

Updated. Thanks.